tensorflow/core/platform/env.cc

```diff
@@ -481,15 +486,26 @@ Status FileSystemCopyFile(FileSystem* src_fs, const string& src,
 
   uint64 offset = 0;
   std::unique_ptr<char[]> scratch(new char[kCopyFileBufferSize]);
-  Status s = Status::OK();
-  while (s.ok()) {
+  while (file_size > 0) {
     StringPiece result;
-    s = src_file->Read(offset, kCopyFileBufferSize, &result, scratch.get());
-    if (!(s.ok() || s.code() == error::OUT_OF_RANGE)) {
+    size_t buffer_size = kCopyFileBufferSize;
+    if (file_size < kCopyFileBufferSize) {
```

Comment:
```suggestion
    if (file_size < buffer_size) {
```

Reply:
@mihaimaruseac 
Thanks for the comments, your suggestion is very good, this PR is a repair to HDFS. Need to add a judgment for different file systems. But I will suspend this PR for a while, wait for the conclusion of [patch2](https://github.com/tensorflow/tensorflow/pull/42599), and choose one of the two

Reply:
@mihaimaruseac Can you help review the code？ In the HDFS scenario, the interface for copying files is adapted to avoid triggering reopen.